err_stash.py

```diff
@@ -42,22 +46,82 @@ def fetch_repo_commits(self, project, slug, until, since):
         return self._stash.projects[project].repos[slug].commits(until, since)
 
 
+class GithubAPI:
+    """
+    Access to the pygithub API.
+    """
+    def __init__(self, login_or_token=None, password=None):
+        self._github = Github(login_or_token=login_or_token, password=password)
+        self.url = 'https://github.com'
+        # disable PyGithub logger
+        logging.disable(logging.CRITICAL)
+
+    def fetch_repos(self, organization:str):
+        return list(self._github.get_organization(organization).get_repos())
+
+    def fetch_branches(self, organization:str, repo_name:str, *, branch_name:str=''):
+        repo = self._github.get_organization(organization).get_repo(repo_name)
+        if branch_name == '':
+            return list(repo.get_branches())
+
+        try:
+            return [repo.get_branch(branch_name)]
+        except GithubException as e:
+            if e.status == 404:  # branch doesn't exist on this repo
+                return []
+            else:
+                raise
+
+    def delete_branch(self, organization:str, repo_name:str, branch_name:str, pr_id):
+        """
+        Deletes the branch based on the pr_id, then Github closes the PR.
+        """
+        repo = self._github.get_organization(organization).get_repo(repo_name)
+        git_ref = repo.get_git_ref("heads/{ref}".format(ref=repo.get_pull(pr_id).head.ref))
+        assert branch_name in git_ref.ref, "Trying to delete the wrong branch, check the PR ID."
+        git_ref.delete()
+
+    def fetch_pull_requests(self, organization, repo_name):
+        """
+        Returns a list with all open pull requests
+        """
+        return list(self._github.get_organization(organization).get_repo(repo_name).get_pulls())
+
+    def fetch_pull_request(self, organization, repo_name, pr_id):
+        return self._github.get_organization(organization).get_repo(repo_name).get_pull(pr_id)
+
+    def fetch_repo_commits(self, organization, repo_name, from_branch:str, to_branch:str):
+        """
+        Returns a list of commits that are on from_branch, but not in to_branch, i.e.,
+        commits that are added by the PR.
+
+        :param from_branch:
+             Name of the branch that created the PR (Github calls this head branch).
+
+        :param to_branch:
+            Name of the target branch in the PR (Github calls this base branch).
+        """
+        repo = self._github.get_organization(organization).get_repo(repo_name)
+        return repo.compare(to_branch, from_branch).commits
+
+
 class MergePlan:
     """
     Contains information about branch and PRs that will be involved in a merge operation.
     """
 
-    def __init__(self, project, slug):
+    def __init__(self, project, slug, *, comes_from_github=False):
```

Comment:
Is not a `source` param better than a flag `comes_from_github`?

Reply:
I'd say a `source:ApiType` where `ApiType` is an `Enum` would be nice if we had more than 2 APIs, but I think we won't need that anytime soon.

Reply:
Ok... but I can hear Bruno complaining about this =P